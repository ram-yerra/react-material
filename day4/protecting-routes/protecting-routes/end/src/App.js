import React, { useState } from "react";
import Home from "./components/Home";
import Books from "./components/Books";
import Secret from "./components/Secret";
import UserCtx from "./components/UserCtx";
import Login from "./components/Login";
import Protected from "./components/Protected";
import { BrowserRouter, Switch, Route, Link } from "react-router-dom";

const App = () => {
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  return (
    <BrowserRouter>
      <UserCtx.Provider
        value={{
          isLoggedIn,
          doLogin: code =>
            code === "pizza" ? setIsLoggedIn(true) : setIsLoggedIn(false)
        }}
      >
        <div className="app">
          <h1>Book Store</h1>
          <div className="navigation">
            <Link to="/" id="home-link">
              Home
            </Link>
            <Link to="/books" id="books-link">
              Books
            </Link>
            <Link to="/secret" id="secret-link">
              Secret Collection
            </Link>
          </div>
          <div className="content">
            <Switch>
              <Route exact path="/">
                <Home />
              </Route>
              <Route path="/books/:id?">
                <Books />
              </Route>
              <Protected isLoggedIn={isLoggedIn} path="/secret">
                <Secret />
              </Protected>
              <Route path="/login">
                <Login />
              </Route>
            </Switch>
          </div>
        </div>
      </UserCtx.Provider>
    </BrowserRouter>
  );
};

export default App;
