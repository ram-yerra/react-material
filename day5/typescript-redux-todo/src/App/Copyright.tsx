import React from 'react'

const Copyright: React.FC = () => (
  <footer className="info">
    <p>TodoMVC</p>
  </footer>
)

export default Copyright
