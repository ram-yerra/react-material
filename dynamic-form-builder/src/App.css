.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 60px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  font-size: 1rem;
  padding: 0 20px;
  color: white;
}

.App-link {
  color: #61dafb;
}

.App-content {
  padding: 20px;
}
