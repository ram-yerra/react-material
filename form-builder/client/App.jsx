import React, {Suspense} from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';

import Racer from 'racer';
import orm from '@startupjs/orm';
import { model, initLocalCollection } from '@startupjs/sharedb'
import init from '@startupjs/init/lib/index.web';

import 'bootstrap/dist/css/bootstrap.min.css';
import './stylus/mixins.styl';
import './stylus/styles.styl';
import Header from './modules/common/components/header/Header';
import Loading from './modules/common/components/loading/Loading';

const FormsList = React.lazy(() => import('./modules/forms/components/formsList/FormsList'));
const FormDetails = React.lazy(() => import('./modules/forms/components/formDetails/FormDetails'));
const AddEditForm = React.lazy(() => import('./modules/forms/components/addEditForm/AddEditForm'));
const About = React.lazy(() => import('./modules/about/components/About'));

initLocalCollection('_about')
initLocalCollection('_forms')
initLocalCollection('_fields')

const baseUrl = 'http://localhost:8080';
init({ baseUrl})

model.connection.send = function send(message) {
  getToken().then(myToken => {
    message.token = myToken
    this.emit('send', message)
    this.socket.send(JSON.stringify(message))
  })
};

const getToken = () => {
  return new Promise((resolve, reject) => {
    resolve('sampletoken');
  })
};

// Init orm of @startupjs with racer
Racer.use(orm);
// Init ORM for modules
require('../loaders/modelLoader.js?context=client!').forEach(m => Racer.use(m))
Racer.use(commonModel)

function App() {
  return (    
      <div className='app'>
        <Header />
        <div className='app-content'> 
          <Suspense fallback={<Loading />}>
            <Switch>
              <Route exact path='/'>
                <Redirect to="/forms" />  
              </Route>          
              <Route path='/forms/:id' component={FormDetails}></Route>
              <Route match path='/forms' component={FormsList}></Route> 
              <Route path='/forms/add' component={AddEditForm}></Route>          
              <Route path='/about' component={About}></Route>          
            </Switch>
          </Suspense>
        </div>
      </div>
  );
}

export default App;
