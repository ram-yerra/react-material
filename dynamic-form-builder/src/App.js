import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';

import './App.css';
import FormsList from './components/formsList/FormsList';
import About from './components/about/About';
import Header from './components/header/Header';
import FormDetails from './components/formDetails/FormDetails';
import AddEditForm from './components/addEditForm/AddEditForm';

function App() {
  return (
    <div className='App'>
      <Header />
      <div className='App-content'> 
        <Switch>
          <Route exact path='/'>
            <Redirect to="/forms" />  
          </Route>          
          <Route path='/forms/:id' component={FormDetails}></Route>
          <Route path='/forms' component={FormsList}></Route> 
          <Route path='/forms/add' component={AddEditForm}></Route>          
          <Route path='/about' component={About}></Route>          
        </Switch>
      </div>
    </div>
  );
}

export default App;
