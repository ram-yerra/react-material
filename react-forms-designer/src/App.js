import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Route, Switch } from 'react-router-dom';

import 'bootstrap/dist/css/bootstrap.css';
import './App.scss';

import FormDesigner from './components/formDesigner/FormDesigner';
import FormPreview from './components/formPreview/FormPreview';

import { FormFieldsProvider } from './contexts/FormFieldsContext';

function App() {
  return (
    <div className="App container-fluid">
        <Router>      
              <Switch>
                  <FormFieldsProvider>
                    <Route exact path="/" component={FormDesigner}></Route>
                    <Route path="/preview-form" component={FormPreview}></Route>
                  </FormFieldsProvider>
              </Switch>
        </Router>
    </div>
  );
}

export default App;
