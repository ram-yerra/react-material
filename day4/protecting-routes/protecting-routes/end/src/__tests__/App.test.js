import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act, Simulate } from "react-dom/test-utils";

import App from "../App";

let div = null;
beforeEach(() => {
  div = document.createElement("div");
  document.body.appendChild(div);
});

afterEach(() => {
  unmountComponentAtNode(div);
  div.remove();
  div = null;
});

describe("Protected Routes", () => {
  it("Route protection was implemented & is working as expected!", async () => {
    await act(async () => render(<App />, div));
    const secretLink = document.querySelector("#secret-link");
    expect(div.innerHTML).toMatchSnapshot();
    await act(async () => {
      secretLink.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    expect(div.innerHTML).toMatchSnapshot();
    const passcodeInput = document.querySelector("#login-passcode");
    passcodeInput.value = "pizza";
    await act(async () => {
      await passcodeInput.dispatchEvent(
        new KeyboardEvent("keyup", { key: "Enter", keyCode: 13, bubbles: true })
      );
    });
    expect(div.innerHTML).toMatchSnapshot();
  });
});
