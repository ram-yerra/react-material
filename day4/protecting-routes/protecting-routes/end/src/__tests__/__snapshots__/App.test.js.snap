// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Protected Routes Route protection was implemented & is working as expected! 1`] = `"<div class=\\"app\\"><h1>Book Store</h1><div class=\\"navigation\\"><a id=\\"home-link\\" href=\\"/\\">Home</a><a id=\\"books-link\\" href=\\"/books\\">Books</a><a id=\\"secret-link\\" href=\\"/secret\\">Secret Collection</a></div><div class=\\"content\\"><div class=\\"home\\"><h3>Welcome to the Book Store app! You can browse our collection on the Books page</h3></div></div></div>"`;

exports[`Protected Routes Route protection was implemented & is working as expected! 2`] = `"<div class=\\"app\\"><h1>Book Store</h1><div class=\\"navigation\\"><a id=\\"home-link\\" href=\\"/\\">Home</a><a id=\\"books-link\\" href=\\"/books\\">Books</a><a id=\\"secret-link\\" href=\\"/secret\\">Secret Collection</a></div><div class=\\"content\\"><div class=\\"login\\"><input type=\\"password\\" id=\\"login-passcode\\" placeholder=\\"Enter password &amp; press enter...\\"></div></div></div>"`;

exports[`Protected Routes Route protection was implemented & is working as expected! 3`] = `"<div class=\\"app\\"><h1>Book Store</h1><div class=\\"navigation\\"><a id=\\"home-link\\" href=\\"/\\">Home</a><a id=\\"books-link\\" href=\\"/books\\">Books</a><a id=\\"secret-link\\" href=\\"/secret\\">Secret Collection</a></div><div class=\\"content\\"><div class=\\"secret\\"><h3>Secret Book of the Week</h3><div class=\\"book-details\\"><img class=\\"bd-thumb\\" src=\\"https://s3.amazonaws.com/AKIAJC5RLADLUMVRPFDQ.book-thumb-images/koskela2.jpg\\" alt=\\"Effective Unit Testing\\"><div class=\\"bd-title\\">Effective Unit Testing</div><div class=\\"bd-authors\\">Lasse Koskela</div><div class=\\"bd-isbn\\">ISBN: 1935182579</div><div class=\\"bd-pg-count\\">Pages: 350</div><div class=\\"bd-short-desc\\"></div></div></div></div></div>"`;
